### ISSUE TITLE
Can xremap work with bouncekeys? Is there a workaround?

### ISSUE THREAD

**Issue by berwynhoyt** [xremap/xremap#563]
TLDR: xremap doesn't always work with bouncekeys. Can you make suggestions for a workaround? Bouncekeys is designed to suppress fast duplicate keypresses (like spacebars sometimes accidentally emit (in my case), or people with nervous conditions). I believe the problem is that an xremap output sequence like `[C-f, C-b, home]` produces two presses of Ctrl in rapid succession so that bouncekeys, unfortunately, suppresses the second Ctrl, making that sequence emit `[C-f, b, home]` instead.

Detail:
* Various operating systems have a workaround for duplicate accidental keys called "bouncekeys" in XKB. I access it in Linux Mint `Menu -> Accessibility -> Typing Assistance`.
* Bouncekeys makes the OS ignore duplicate keys when the key is repeated too quickly.
* The following is a minimal reproduction of the issue, cut down from my [Wordstar keymap for xremap](https://github.com/berwynhoyt/wordstar-keys/tree/main/xremap).

Here is the `xremap` YML file that produces the problem:
```
virtual_modifiers:
  - CapsLock

keymap:
  - name: main
    remap:
      CapsLock-s: [C-b, C-f, home]
```

It is designed to output the key sequence [C-b, C-f, home] whenever `CapsLock-s` is pressed. (That sequence moves the cursor to the start of the bash command line, assuming default bash key bindings).

* If bouncekeys is turned on BEFORE xremap runs, then (oddly) everything works fine.
* If bouncekeys is turned on AFTER xremap runs, the output sequence is [C-b, f, home] instead of [C-b, C-f, home].

The following log is produced by xremap. It all looks normal, but gives a clue as to the problem:
```
$ RUST_LOG=debug xremap bouncekeys.yml
...
/dev/input/event2 : AT Translated Set 2 keyboard
------------------------------------------------------------------------------
[2024-12-21T10:56:20Z DEBUG xremap::event_handler] => 0: KEY_ENTER
[2024-12-21T10:56:20Z DEBUG xremap::action_dispatcher] 0: KEY_ENTER
[2024-12-21T10:56:29Z DEBUG xremap::event_handler] => 1: KEY_CAPSLOCK
[2024-12-21T10:56:29Z DEBUG xremap::event_handler] => 1: KEY_S
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 1: KEY_LEFTCTRL
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 1: KEY_B
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 0: KEY_B
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 0: KEY_LEFTCTRL
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 1: KEY_LEFTCTRL
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 1: KEY_F
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 0: KEY_F
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 0: KEY_LEFTCTRL
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 1: KEY_HOME
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 0: KEY_HOME
[2024-12-21T10:56:29Z DEBUG xremap::event_handler] => 0: KEY_S
[2024-12-21T10:56:29Z DEBUG xremap::action_dispatcher] 0: KEY_S
[2024-12-21T10:56:29Z DEBUG xremap::event_handler] => 0: KEY_CAPSLOCK
```

I think the problem is that, as you can see above, the Ctrl key is pressed twice in rapid succession. Apparently bouncekeys therefore suppresses the second press of Ctrl, so the key that comes out is `f` rather than `C-f`.

The practical effect is that my Wordstar key map doesn't work with bouncekeys.

Can you think of any way around this? For example:
* Is there a setting in xremap that presses Ctrl only once for multiple keys and doesn't lift it after every key?
* I can almost implement bouncekeys (for the spacebar) myself, within xremap. I'd use set-mode when space is first pressed, and switch to a mode where space outputs nothing. But there would need to be a mode timeout back to default mode so that spaces work again after the bounce delay.

Any ideas for a fix?

**Comment by wistoft** [comment]
Hi. I think the easiest way is to have a sleep in the your remap. I have made a PR #574. Your config would become

```yml
virtual_modifiers:
  - CapsLock

keymap:
  - name: main
    remap:
      CapsLock-s: [C-b, { sleep: 100 }, C-f, home]
```

It is also possible to implement the feature you mention in #571. But that requires a little more work to think through possible corner cases.

A general solution to bounce keys should probably be activated in xremap, not in LM. Because xremap is handling keys first, and must be able to ignore keys and avoid emitting a remap twice.

> Is there a setting in xremap that presses Ctrl only once for multiple keys and doesn't lift it after every key?

There isn't. Whether it's complex to implement, I don't know.

**Comment by berwynhoyt** [comment]
> Hi. I think the easiest way is to have a sleep in the your remap. I have made a PR #574. Your config would become

@wistoft, thanks for #574. I'll try it when I get back home on Friday. It looks like a good addition to xremap for a number of situations.

**Comment by k0kubun** [comment]
#574 seems like a nice workaround. Thank you!
